src/groupeddataframe/groupeddataframe.jl

```diff
@@ -460,6 +462,19 @@ true
 """
 Base.keys(gd::GroupedDataFrame) = GroupKeys(gd)
 
+# assume that GroupKey was generated so that we do not have to check the index
+Base.haskey(gd::GroupedDataFrame, key::GroupKey) = gd === parent(key)
+Base.haskey(gd::GroupedDataFrame, key::Tuple) = haskey(gd.keymap, key)
+
+function Base.haskey(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
+    if length(key) != length(gd.cols) || any(n != _names(gd)[c] for (n, c) in zip(N, gd.cols))
```

Comment:
Design question: is it more useful to return `false` than throwing an error when the number or names of columns doesn't match? Throwing could help catch errors, and I'm not sure I can imagine cases where it would be useful to return `false` instead.

Reply:
Good point. I will change it.